### DISCUSSION TITLE
[Scenario Editor] It should not be possible to authorize a magic book without a spell

### DISCUSSION THREAD

**Discussion by LeHerosInconnu** [ihhub/fheroes2#8755] [General]
In fheroes2, it should not be possible to authorize a magic book without a spell.

It doesn't make much sense to allow a hero to own a magic book without a spell.
~This is not possible in the original game either.~
What could be a valid reason?
If the reason is that the hero shouldn't have a spell, then he shouldn't be given a magic book.

In fheroes2:

![Default class hero spell 01](https://github.com/ihhub/fheroes2/assets/43583994/352fc2a8-33aa-40e2-a8e1-e051a1d50215)

By default, the hero's magic book should contain the default spell for the hero type.
- For the Sorceress: Bless
- For the Warlock: Curse
- For the Necromancer: Haste
- For the Wizard: Stoneskin

For the Barbarian, the default spell could be Bloodlust.
For the Knight, the default spell could be Slow.

**Comment by oleg-derevenetz** [comment]
> What could be a valid reason?

The reason is to get the spell book for free instead of 500 gold.

**Reply by zenseii** [reply]
You could design a map where you have to pick up a magic book and can only learn spells from shrines or from eagle eye. Lots of possibilities exist so I don't think we need should force these out of existance.

**Reply by LeHerosInconnu** [reply]
Hello @oleg-derevenetz and @zenseii,

> > What could be a valid reason?
> 
> The reason is to get the spell book for free instead of 500 gold.

The magic book is also free if it contains at least one spell, and it's even better as it already contains a spell.

> You could design a map where you have to pick up a magic book and can only learn spells from shrines or from eagle eye. Lots of possibilities exist so I don't think we need should force these out of existance.

Nothing would prevent the scenario creator to make the hero start without a magic book in this case, and if the hero gets a magic book it should contain at least one spell.
A magic book without a spell is just an empty notebook, not a magic book. :)
I have made a proposal regarding the magic book artifact on the adventure map here: https://github.com/ihhub/fheroes2/discussions/8753.

**Comment by LeHerosInconnu** [comment]
Let's go with an empty magic book for the hero.
I close the discussion.
